pyth/tests/test_publish.py

```diff
@@ -19,6 +29,59 @@ def get_price_acct():
         output = output.decode('ascii')
         output = json.loads(output)
         return output['price_accounts'][0]
+    
+    def get_key_pair(path_to_file):
+        with  open(path_to_file) as key_file:
+            key_data = json.load(key_file)
+        return Keypair.from_secret_key(key_data)
+    
+    def get_path_to_pythdir_pair(account_addr):
+        return os.path.join(pyth_dir, "account_" + str(account_addr) + ".json")
+
+
+    
+    def resize_account(price_account_address):
+        """
+        given a string with the pubkey of a price accountm it calls the resize instruction of the Oracle on it
+        """
+        #constants from oracle.h
+        PROGRAM_VERSION = 2 #TODO: update this
+        COMMAND_UPD_ACCOUNT = 14
+        SYSTEM_PROGRAM = "11111111111111111111111111111111"
+
+        #update version of price accounts to make sure they resize
+        layout = Struct("version" / Int32ul, "command" / Int32sl)
+        data = layout.build(dict(version=PROGRAM_VERSION, command=COMMAND_UPD_ACCOUNT))
+        funding_key = PublicKey(solana_keygen[0])
+        price_key = PublicKey(price_account_address)
+        system_key = PublicKey(SYSTEM_PROGRAM)
+        resize_instruction = TransactionInstruction(
+            data=data,
+            keys=[
+                AccountMeta(pubkey=funding_key, is_signer=True, is_writable=True),
+                AccountMeta(pubkey=price_key, is_signer=True, is_writable=True),
+                AccountMeta(pubkey=system_key, is_signer=False, is_writable=False),
+            ],
+            program_id = PublicKey(solana_program_deploy),
+        )
+        txn = Transaction().add(resize_instruction)
+        txn.fee_payer = funding_key
+        sender = get_key_pair(solana_keygen[1])
```

Comment:
Nickpick call this `funding_key_pair`.

Reply:
Not really, the type is PublicKey.

Reply:
It's a signer to it has to be a keypair

Reply:
No, the signing happens somehwere else :)

Reply:
`solana_client.send_transaction(txn, sender, price_key_pair)`
This is where signing happens, it just needs the public key for the feepayer

Reply:
I can changer sender to fee_payer_keypair, will do in a sec

Reply:
Sorry, my comment was clear : I'm proposing renaming `sender` to `funding_key_pair`